server/app/src/androidTest/java/sh/calaba/instrumentationbackend/StatusReporterObject.java

```diff
@@ -0,0 +1,76 @@
+package sh.calaba.instrumentationbackend;
+
+import android.os.Environment;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+public class StatusReporterObject {
+
+    public static void report(String method,
+                              String message,
+                              StatusReporter.FinishedState extraState,
+                              boolean hasData) {
```

Comment:
I can't see any cases where this parameter may be equal to `!true`. Why do we need it and further `if` statement?

Reply:
Also legacy. Copied from activity where it was needed (Intent could have no data), but there is no need anymore as long as we call the method directly